libvmaf/src/feature/cambi.c

```diff
@@ -719,9 +734,34 @@ static FORCE_INLINE inline double weight_scores_per_scale(double *scores_per_sca
     return score / normalization;
 }
 
-static int cambi_score(VmafPicture *pics, uint32_t *mask_dp, uint16_t window_size, double topk,
-                       const uint16_t *tvi_for_diff, float *c_values, uint16_t *c_values_histograms, 
-                       uint8_t *filter_mode_histogram, uint16_t *filter_mode_buffer, double *score) {
+static void write_uint16(FILE *file, uint16_t v) {
+    fwrite((void*)(&v), sizeof(v), 1, file);
+}
+
+static void dump_c_values(const float *c_values, int width, int height, int scale, int window_size, const char *dir_path) {
+    int max_diff_weight = g_diffs_weights[0];
+    for (int i = 0; i < NUM_DIFFS; i++) {
+        if (g_diffs_weights[i] > max_diff_weight) {
+            max_diff_weight = g_diffs_weights[i];
+        }
+    }
+    char path[1024];
+    snprintf(path, sizeof(path), "%s/cambi_heatmap_scale_%d_%dx%d_16b.gray", dir_path, scale, width, height);
+    int max_c_value = max_diff_weight * window_size * window_size / 4;
+    int max_16bit_value = (1 << 16) - 1;
+    double scaling_value = (double)max_16bit_value / max_c_value;
+    FILE *file = fopen(path, "a");
```

Comment:
Your call to `fopen()` can fail, add a check here and bail if it's no good.

Reply:
By "bail" you mean abort the whole program, right? I would say that continuing execution without writing to disk is not desirable.

Reply:
No, just return an error and handle it somehow. If fopen fails and your file pointer is NULL you will get a SIGSEGV.